src/Compilers/Core/Portable/Operations/Statement.cs

```diff
@@ -1,13 +1,20 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Immutable;
+
 namespace Microsoft.CodeAnalysis.Semantics
 {
     internal partial class VariableDeclaration : IVariableDeclaration
     {
         public VariableDeclaration(ILocalSymbol variable, IOperation initialValue, SyntaxNode syntax) :
-            this(variable,
+            this(ImmutableArray.Create(variable), initialValue, syntax)
+        {
+        }
+
+        public VariableDeclaration(ImmutableArray<ILocalSymbol> variables, IOperation initialValue, SyntaxNode syntax) :
+            this(variables,
                 initialValue,
-                variable == null || (initialValue != null && initialValue.IsInvalid),
+                variables.IsDefault || (initialValue != null && initialValue.IsInvalid),
```

Comment:
I don't believe an empty set is possible here. Even just `Dim` has a non-empty list of declarations. I'll make this `IsDefaultOrEmpty` and add a test for just `Dim` and just `int` (just `var` turns into just an `IExpressionStatement`).